### PR TITLE
Fix cifar10 model

### PR DESCRIPTION
**Pull request by joost-ve**
- Replaces dependencies which prevented updating to flux 0.11.4
- Wrote custom training loop to allow data to be moved to the GPU in batches, this allows running the model on lower end cards without OOM errors.

### CODE REVIEW

File: vision/cifar10/cifar10.jl [hunk 1]
```diff
@@ -158,21 +145,49 @@ function train(; kws...)
...
+
+            gs = Flux.gradient(ps) do 
+                loss(x, y)
+            end
+
+            Flux.update!(opt, ps, gs)
+        end
+
+        validation_loss = 0f0
+        for (x, y) in val_data
```
Comment (DhairyaLGandhi) [inline vision/cifar10/cifar10.jl:166]: Validation should also be batched I think

Comment (joost-ve) [reply vision/cifar10/cifar10.jl:166]: What do you mean? val_data is a batched data loader, same as train_data, see line 139.

File: vision/cifar10/cifar10.jl [hunk 2]
```diff
@@ -20,40 +20,26 @@ end
...
+    val_x = float(val_x)
+    val_y = onehotbatch(val_y, 0:9)
+    
+    return (train_x, train_y), (val_x, val_y)
+end
 
-    test = (testX,testY)
-    return test
+function get_test_data()
+    test_x, test_y = CIFAR10.traindata()
```
Comment (CarloLucibello) [inline vision/cifar10/cifar10.jl:37]:
```suggestion
    test_x, test_y = CIFAR10.testdata()
```

File: vision/cifar10/cifar10.jl [hunk 3]
```diff
@@ -158,21 +145,49 @@ function train(; kws...)
...
+    # Defining the optimizer
     opt = ADAM(args.lr)
+    ps = Flux.params(m)
+
     @info("Training....")
     # Starting to train models
-    Flux.@epochs args.epochs Flux.train!(loss, params(m), train, opt, cb = evalcb)
+    for epoch in 1:args.epochs
+        for (x, y) in train_data
+            x, y = gpu(x), gpu(y)
```
Comment (CarloLucibello) [inline vision/cifar10/cifar10.jl:156]: define a `device` instead of using `gpu`?

File: vision/cifar10/cifar10.jl (hunk 2 above)
Comment (joost-ve) [reply vision/cifar10/cifar10.jl]: Oops :wink:

File: vision/cifar10/cifar10.jl [hunk 4]
```diff
@@ -20,159 +20,185 @@ end
...
+
+            Flux.update!(opt, ps, gs)
+        end
+
+        validation_loss = 0f0
+        for (x, y) in val_data
+            x, y = x |> device, y |> device
+            validation_loss += loss(x, y)
+        end
+        @show validation_loss
```
Comment (CarloLucibello) [inline vision/cifar10/cifar10.jl:180]:
```suggestion
         validation_loss /= length(val_data)
        @show validation_loss
```

**Pull request merged**